satpy/readers/netcdf_utils.py

```diff
@@ -143,13 +146,15 @@ def collect_metadata(self, name, obj):
         # Look through each subgroup
         base_name = name + "/" if name else ""
         for group_name, group_obj in obj.groups.items():
-            self.collect_metadata(base_name + group_name, group_obj)
+            full_group_name = base_name + group_name
+            self.file_content[full_group_name] = group_obj
+            self._collect_attrs(full_group_name, group_obj)
+            self.collect_metadata(full_group_name, group_obj)
```

Comment:
@gerritholl you worked with this last, would you mind checking this change?

Reply:
Looks good to me.  I originally introduced this for performance reasons, I observe no performance difference in the script I used to measure this back then.

Reply:
Thanks for checking it out